### PR TITLE
ENH: Adds `discard_untrimmed` to trim methods

### LINKED ISSUE #10: trim-*: discard unmatched
This cutadapt parameter controls if unmatched reads should be discarded - this would be pretty useful to wrap (and straightforward).

This recently [came up on the forum](https://forum.qiime2.org/t/import-multiplexed-r1-fastq-and-r2-fastq-with-mixed-forward-and-reverse-reads-truncate-reverse-primer/2052/17?u=thermokarst).

### PR DESCRIPTION
**Pull request by thermokarst**
Fixes #10 

This still needs a tire-kicking. :wheel_of_dharma:

### CODE REVIEW

File: q2_cutadapt/tests/test_trim.py [hunk 1]
```diff
@@ -156,7 +156,8 @@ def test_build_trim_command_typical(self):
                                       overlap=4,
                                       match_read_wildcards=True,
                                       match_adapter_wildcards=False,
-                                      minimum_length=2)
+                                      minimum_length=2,
+                                      discard_untrimmed=True)
```
Comment (nbokulich) [inline q2_cutadapt/tests/test_trim.py]: here and below: should we be making sure that this is working as intended? i.e., that specific IDs are dropped/retained (or at least a specific # of seqs is retained). I get it if you don't feel we should be in the business of validating cutadapt...

File: q2_cutadapt/plugin_setup.py [hunk 2]
```diff
@@ -107,6 +108,7 @@
                           'the cutadapt default of 0 has been overridden, '
                           'because that value produces empty sequence '
                           'records.',
+        'discard_untrimmed': 'Discard reads in which an adapter was found.',
```
[the remark this answers is not in the document]
Comment (thermokarst) [reply q2_cutadapt/plugin_setup.py]: Oops, copied the wrong text from the cutadapt docs.

File: q2_cutadapt/plugin_setup.py [hunk 3]
```diff
@@ -206,6 +209,7 @@
...
```
[the remark this answers is not in the document]
Comment (thermokarst) [reply q2_cutadapt/plugin_setup.py]: same as above.

File: q2_cutadapt/tests/test_trim.py (hunk 1 above)
Comment (thermokarst) [reply q2_cutadapt/tests/test_trim.py]: My thinking here is test that we are writing the command out correctly, leave the implementation up to cutadapt. However, given that this whole problem came out of the "empty record" problem, I should probably add at least one test for _that case_. Thanks!

**Pull request merged**